File: src/types/voice.ts
export interface Voice {
  id: string;
  name: string;
  description: string;
  avatar: string;
  language: string;
  category: 'Cinematic' | 'Synthetic' | 'Character' | 'Narrative' | 'Radio';
  isFavorite: boolean;
  stability: number; // 0-100
  similarity: number; // 0-100
  styleExaggeration: number; // 0-100
  sampleUrl?: string;
  gender?: 'Male' | 'Female' | 'Neutral';
  tags: string[];
  createdAt: string;
  providerId?: string; // e.g. elevenlabs, xtts, piper, google
  providerVoiceId?: string;
  
  // Advanced Voice Settings
  accent?: string;
  pitch?: number;
  speed?: number;
  volume?: number;
  emotion?: string;
  pauseDuration?: number;
  pronunciationDictionary?: Record<string, string>;
  speakingStyle?: string;
  voiceNotes?: string;
  age?: 'Child' | 'Young Adult' | 'Adult' | 'Senior';
}

export type VoiceViewMode = 'grid' | 'list';
export type VoiceFilterTab = 'all' | 'favorites' | 'recent';

File: backend/app/schemas/voice.py
from pydantic import BaseModel
from typing import Optional, List

class VoiceBase(BaseModel):
    name: str
    description: Optional[str] = None
    providerId: str
    providerVoiceId: Optional[str] = None
    language: str = "en"
    gender: Optional[str] = None
    tags: List[str] = []

class VoiceCreate(VoiceBase):
    pass

class VoiceResponse(VoiceBase):
    id: str

    class Config:
        from_attributes = True
